lib/cypress_on_rails/vcr_wrapper.rb

```diff
@@ -0,0 +1,38 @@
+require 'rack'
+require 'cypress_on_rails/configuration'
+
+module CypressOnRails
+  class VCRWrapper
+
+    def initialize(app:, env:)
+      @app = app
+      @env = env
+      @request = Rack::Request.new(env)
+    end
+
+    def run_with_cassette
+      VCR.use_cassette(cassette_name, { :record => configuration.vcr_record_mode }) do
+        logger.info "Handle request with cassette name: #{cassette_name}"
+        @app.call(@env)
+      end
+    end
+
+    private
+
+    def configuration
+      CypressOnRails.configuration
+    end
+
+    def logger
+      configuration.logger
+    end
+
+    def cassette_name
```

Comment:
At this point, the most tricky part is `cassette_name`.
It should define cassette name based on some data from the request. But to do this right we need to be careful.
I'm not sure we could pass this name from Cypress, b/c BE request could be sent from anywhere (ex. graphql mutation in form submit)

Reply:
For those projects using GraphQL it would be nice to detect it as well. Right now it considers `/graphql` as a path for endpoint, but it could be different. 
Also, no guarantee, that the operation name will be a part of `params`